### ISSUE TITLE
add documentation about correctly refreshing expiring JWT's using the refresh token

### ISSUE THREAD

**Issue by fool** [netlify/netlify-identity-widget#117]
duplicate of https://github.com/netlify/gotrue-js/issues/20 intentionally - both docs need updating cc @futuregerald @biilmann

**Comment by skllpr** [comment]
Definitely something that should have happened in 2 years.

**Comment by michalhudecek** [comment]
Yes, please! This is really needed. It is not clear at all how exactly and where to use:

```
netlifyIdentity.refresh().then((jwt)=>console.log(jwt))
```

When exactly should we trigger the above code? `triggerNetlifyIdentityAction('login')` is called only on user click on login / logout button which means the tokens are not refreshed automatically on page load before calling Netlify lambda functions that require login.

What should we actually do with the response instead of just the console.log? Should we store it back to currentUser.access_token or currentUser.refresh_token?

**Comment by michalhudecek** [comment]
The Netlify Identity Widget is completely useless when it keeps loggin out users after one hour.

**Comment by SamuelToh** [comment]
Should the api also be returning the new expiry time?

**Comment by tam-carre** [comment]
> The Netlify Identity Widget is completely useless when it keeps logging out users after one hour.

This is exactly what I'm experiencing, and I can't find a fix. Can you help me?

**Comment by fool** [comment]
The widget doesn't cause logout after 1 hour;  Netlify's system does (so it will happen even if you don't use the widget).  There is no way to change that timeout;  you will need to ensure the token is up to date when any request is being made from your front-end by using the code that was suggested before (from your front-end, before making any other requests for content, to your netlify-hosted site).

**Comment by alucidwolf** [comment]
Has anyone come up with a solution for this?

Regardless of front end framework, some use of navigation guards or routing to refresh the token has not been successful from my end to just force the `access_token` in local storage to use the new `JWT` from the refresh response.

The 1-hour expiry for the user being logged in and then kicked out is not the desired user experience for many situations.

The JWT we get back from the `netlifyIdentity.refresh()` promise is just the same `access_token` string and not a new value for the `refresh_token` key on the `token` object in local storage for the `gotrue.user` key

**Comment by HideoKun** [comment]
> Has anyone come up with a solution for this?
> 
> Regardless of front end framework, some use of navigation guards or routing to refresh the token has not been successful from my end to just force the `access_token` in local storage to use the new `JWT` from the refresh response.
> 
> The 1-hour expiry for the user being logged in and then kicked out is not the desired user experience for many situations.
> 
> The JWT we get back from the `netlifyIdentity.refresh()` promise is just the same `access_token` string and not a new value for the `refresh_token` key on the `token` object in local storage for the `gotrue.user` key

Any updates? This one is also down for me :(
